src/wallet/rpcwallet.cpp

```diff
@@ -3183,8 +3180,8 @@ UniValue fundrawtransaction(const JSONRPCRequest& request)
             if (options.exists("changeAddress")) {
                 throw JSONRPCError(RPC_INVALID_PARAMETER, "Cannot specify both changeAddress and address_type options");
             }
-            coinControl.m_change_type = ParseOutputType(options["change_type"].get_str(), pwallet->m_default_change_type);
-            if (coinControl.m_change_type == OutputType::NONE) {
+            coinControl.m_change_type = pwallet->m_default_change_type;
```

Comment:
Why is this assignment needed? Either `ParseOutputType` in the next line succeeds and overwrites the value or it fails and we get thrown out of this function.

I think you can safely remove this line.

Reply:
Right, also got confused because forgot that `CCoinControl::m_change_type` is `boost::optional<OutputType>`.

Nit, to avoid the *assignment is needed, but the value doesn't matter* detail:
```cpp
OutputType change_type;
if (!ParseOutputType(options["change_type"].get_str(), change_type) {
    ...
}
coinControl.m_change_type = change_type;
```
Otherwise a comment above could help.

Reply:
@ryanofsky I think it's perfectly legal to dereference a pointer to an uninitialized value, as long as you don't read it? This is difference for dereferencing a null pointer, which is always undefined.